### PR TITLE
Draw qubit reset and postselection with matplotlib

### PR DESCRIPTION
**Pull request by albi3ro**
Screenshot 2023-11-13 at 2 36 22 PM

This PR accomplishes two main things:

* Indicates postselection on the icon for mid circuit measurements
* Indicates reset with erasing the wire line and restarting it after a `|0>` box.

There is a regression in the handling of conditionals, as drawing conditionals will be standardized in a follow-up PR.

While doing so, I also:

* converted from a hacky single dispatch to `functools.singledispatch` to register custom ways to display operations
* Moved `Conditional` to the `ops.op_math` submodule to avoid a circular dependency. This can be accomplished in a separate PR instead.  

[sc-50118]

### CODE REVIEW

**Comment by trbromley** [comment]
Thanks @albi3ro, this looks great! One minor suggestion, which could be done in a separate PR: could the measurement symbol (semicircle and arrow) be moved downwards inside the box so that it is centre-aligned vertically?

**Comment by albi3ro** [comment]
@trbromley like this?


Screenshot 2023-11-14 at 10 11 10 AM

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -106,6 +106,8 @@
 * Specifying `control_values` passed to `qml.ctrl` as a string is no longer supported.
   [(#4816)](https://github.com/PennyLaneAI/pennylane/pull/4816)
 
+* The `Conditional` operator is moved from `transforms` to `ops.op_math` to prevent circular import issues.
+
```
Comment (albi3ro) [inline doc/releases/changelog-dev.md:110]:
```suggestion
```

**Review by mudit2812 — COMMENTED** [review]
Looks great Christina!

image

Is it possible to lower the number drawn for postselection? Imo it's too close to the measurement symbol. Other than than, I'm ready to approve.

**Review by mudit2812 — APPROVED** [review]
The link to this PR still needs to be added to the changelog, but other than that looks good! Thanks Christina, the singledispatch was a really good change as well

**Comment by timmysilv** [comment]
what is the plan for conditionals? are we going to restore them to the current behaviour, or do something all-new? not entirely sure why it should be done in another PR, but i'm fine to keep things separated with a plan laid out

**Comment by albi3ro** [comment]
> what is the plan for conditionals? are we going to restore them to the current behaviour, or do something all-new? not entirely sure why it should be done in another PR, but i'm fine to keep things separated with a plan laid out

PR #4850

**Review by timmysilv — APPROVED** [review]
looks good!

**Pull request merged**